bpf/flows.c

```diff
@@ -0,0 +1,79 @@
+#include <vmlinux.h>
+#include <bpf_helpers.h>
+
+#define TC_ACT_OK 0
+#define TC_ACT_SHOT 2
+
+// definitions to not having to include arpa/inet.h version of 32-bit in the compiler
+// TODO: to unload even more the Kernel space, we can do this calculation in the Go userspace
+#define htons(x)                       \
+    ((u16)((((u16)(x)&0xff00U) >> 8) | \
+           (((u16)(x)&0x00ffU) << 8)))
+
+#define htonl(x)                            \
+    ((u32)((((u32)(x)&0xff000000U) >> 24) | \
+           (((u32)(x)&0x00ff0000U) >> 8) |  \
+           (((u32)(x)&0x0000ff00U) << 8) |  \
+           (((u32)(x)&0x000000ffU) << 24)))
+
+struct {
+    __uint(type, BPF_MAP_TYPE_RINGBUF);
+    __uint(max_entries, 1 << 24);
+} flows SEC(".maps");
+
+struct egress_t {
```

Comment:
My understanding is that this struct and the go struct RawStats have to be equivalent.

Anyway we could generate one from the other?

Reply:
Unfortunately, I don't know any way (unless we create some kind of parser, which it would be a nice side project)